### PR TITLE
Support for RSA-PSS signatures

### PR DESCRIPTION
**Pull request by babelouest**
Disclaimer:
- Works only for GnuTLS at that time
- Requires GnuTLS >= 3.6

### CODE REVIEW

File: libjwt/jwt-gnutls.c [hunk 1]
```diff
@@ -191,9 +191,17 @@ int jwt_sign_sha_pem(jwt_t *jwt, char **out, unsigned int *len, const char *str)
 	}
 
 	/* Sign data */
-        if (gnutls_privkey_sign_data2(privkey, alg, 0, &body_dat, &sig_dat)) {
-                ret = EINVAL;
-                goto sign_clean_privkey;
+        if (pk_alg == GNUTLS_PK_EC) {
+                /* ec signatues don't like gnutls_privkey_sign_data2 */
+                if (gnutls_privkey_sign_data(privkey, alg, 0, &body_dat, &sig_dat)) {
```
Comment (benmcollins) [inline libjwt/jwt-gnutls.c]: Question: Does RSA do well with `gnutls_privkey_sign_data()`? If so, can we use that for both?

Comment (babelouest) [reply libjwt/jwt-gnutls.c]: `gnutls_privkey_sign_data` works well with RSA and ECDSA signatures but not with RSA-PSS.
`gnutls_privkey_sign_data2` works well with RSA and RSA-PSS signatures but not with ECDSA.

So we could use `gnutls_privkey_sign_data` with RSA as well if you prefer.

**Comment by benmcollins** [comment]
Hoping someone can resolve the conflicts and possibly add OpenSSL support for RSA-PSS as well.

**Comment by benmcollins** [comment]
RSA-PSS support has been added from outside this PR.

**Pull request closed without merging**